Provider Crashing on Specific Tenant

Hello, we are seeing the provider crash on one of our tenants. We are using the latest `v0.30.0` version of the provider.

### Describe the problem

After running a plan we get the following output:

```
╷
│ Error: Request cancelled
│
│   with auth0_client_grant.user-management-app-client-grant,
│   on main.tf line 74, in resource "auth0_client_grant" "user-management-app-client-grant":
│   74: resource "auth0_client_grant" "user-management-app-client-grant" {
│
│ The plugin.(*GRPCProvider).UpgradeResourceState request was cancelled.
╵

╷
│ Error: Plugin did not respond
│
│   with module.module-auth0-tenant.auth0_branding.brand,
│   on .terraform/modules/module-auth0-tenant/main.tf line 36, in resource "auth0_branding" "brand":
│   36: resource "auth0_branding" "brand" {
│
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more
│ details.
╵

// ... [SNIP] ... many more like the above


Stack trace from the terraform-provider-auth0_v0.30.0 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 121 [running]:
github.com/auth0/terraform-provider-auth0/auth0.flattenClientAddons(0xc000690630)
	github.com/auth0/terraform-provider-auth0/auth0/resource_auth0_client.go:998 +0xbdd
github.com/auth0/terraform-provider-auth0/auth0.readClient({0x19c60a8?, 0xc0001fc8d0?}, 0xc0002fd380, {0x17e4580?, 0xc0006f48c0})
	github.com/auth0/terraform-provider-auth0/auth0/resource_auth0_client.go:642 +0x11ae
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc0000c5c00, {0x19c60a8, 0xc0001fc8d0}, 0xd?, {0x17e4580, 0xc0006f48c0})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.13.0/helper/schema/resource.go:725 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0000c5c00, {0x19c60a8, 0xc0001fc8d0}, 0xc00009d930, {0x17e4580, 0xc0006f48c0})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.13.0/helper/schema/resource.go:1016 +0x47b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc000392600, {0x19c6000?, 0xc0006b8a40?}, 0xc0006b8ac0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.13.0/helper/schema/grpc_provider.go:613 +0x497
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc0003745a0, {0x19c60a8?, 0xc0001fc180?}, 0xc0004ee7e0)
	github.com/hashicorp/terraform-plugin-go@v0.8.0/tfprotov5/tf5server/server.go:746 +0x438
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x186b740?, 0xc0003745a0}, {0x19c60a8, 0xc0001fc180}, 0xc0004ee780, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.8.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000272a80, {0x19c8bd8, 0xc0004824e0}, 0xc00062c000, 0xc0003fe870, 0x1e9ca30, 0x0)
	google.golang.org/grpc@v1.45.0/server.go:1282 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc000272a80, {0x19c8bd8, 0xc0004824e0}, 0xc00062c000, 0x0)
	google.golang.org/grpc@v1.45.0/server.go:1619 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.45.0/server.go:921 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.45.0/server.go:919 +0x28a

Error: The terraform-provider-auth0_v0.30.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

ERRO[0011] 1 error occurred:
	* exit status 1
```

### What was the expected behavior?

The plan would complete without error.

### Reproduction

I believe this is specific to our tenant as the same TF works fine in all of our other tenants except this one. I opened support request `00745742` to look more into the tenant-specific nature of the error and was told to open an issue here.

### Environment

We are using `v0.30.0` of the provider. We are using Terraform `v1.1.2` run via Terragrunt `v0.31.0`

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,10 @@
+## 0.30.1
+
+BUG FIXES:
+
+- `resource/auth0_client`: Fix conversion issue flattenAddons func in client resource ([#140](https://github.com/auth0/terraform-provider-auth0/pull/140))
+
+
 ## 0.30.0
 
 FEATURES:

--- a/auth0/resource_auth0_client.go
+++ b/auth0/resource_auth0_client.go
@@ -995,7 +995,10 @@ func flattenClientAddons(addons map[string]interface{}) []interface{} {
 			"include_attribute_name_format":      samlp["includeAttributeNameFormat"],
 			"binding":                            samlp["binding"],
 			"signing_cert":                       samlp["signingCert"],
-			"logout":                             mapToState(samlp["logout"].(map[string]interface{})),
+		}
+
+		if logout, ok := samlp["logout"].(map[string]interface{}); ok {
+			samlpMap["logout"] = mapToState(logout)
 		}
 
 		m["samlp"] = []interface{}{samlpMap}
@@ -1009,8 +1012,9 @@ func flattenClientAddons(addons map[string]interface{}) []interface{} {
 		"springcm", "wams", "wsfed", "zendesk", "zoom",
 	} {
 		if value, ok := addons[name]; ok {
-			addonType := value.(map[string]interface{})
-			m[name] = mapToState(addonType)
+			if addonType, ok := value.(map[string]interface{}); ok {
+				m[name] = mapToState(addonType)
+			}
 		}
 	}
 

--- a/auth0/resource_auth0_client_test.go
+++ b/auth0/resource_auth0_client_test.go
@@ -86,11 +86,94 @@ func TestAccClient(t *testing.T) {
 					resource.TestCheckResourceAttr("auth0_client.my_client", "signing_keys.#", "1"), // checks that signing_keys is set, and it includes 1 element
 				),
 			},
+			{
+				Config: random.Template(testAccClientConfigWithoutAddonsWithSAMLPLogout, rand),
+				Check: resource.ComposeTestCheckFunc(
+					random.TestCheckResourceAttr("auth0_client.my_client", "name", "Acceptance Test - {{.random}}", rand),
+					resource.TestCheckResourceAttr("auth0_client.my_client", "addons.#", "1"),
+					resource.TestCheckResourceAttr("auth0_client.my_client", "addons.0.samlp.#", "1"),
+					resource.TestCheckResourceAttr("auth0_client.my_client", "addons.0.samlp.0.logout.%", "2"),
+					resource.TestCheckResourceAttr("auth0_client.my_client", "addons.0.samlp.0.logout.callback", "http://example.com/callback"),
+					resource.TestCheckResourceAttr("auth0_client.my_client", "addons.0.samlp.0.logout.slo_enabled", "true"),
+				),
+			},
 		},
 	})
 }
 
 const testAccClientConfig = `
+resource "auth0_client" "my_client" {
+  name = "Acceptance Test - {{.random}}"
+  description = "Test Application Long Description"
+  app_type = "non_interactive"
+  custom_login_page_on = true
+  is_first_party = true
+  is_token_endpoint_ip_header_trusted = true
+  token_endpoint_auth_method = "client_secret_post"
+  oidc_conformant = true
+  callbacks = [ "https://example.com/callback" ]
+  allowed_origins = [ "https://example.com" ]
+  allowed_clients = [ "https://allowed.example.com" ]
+  grant_types = [ "authorization_code", "http://auth0.com/oauth/grant-type/password-realm", "implicit", "password", "refresh_token" ]
+  organization_usage = "deny"
+  organization_require_behavior = "no_prompt"
+  allowed_logout_urls = [ "https://example.com" ]
+  web_origins = [ "https://example.com" ]
+  jwt_configuration {
+    lifetime_in_seconds = 300
+    secret_encoded = true
+    alg = "RS256"
+    scopes = {
+      foo = "bar"
+    }
+  }
+  client_metadata = {
+    foo = "zoo"
+  }
+  addons {
+    firebase = {
+      client_email = "john.doe@example.com"
+      lifetime_in_seconds = 1
+      private_key = "wer"
+      private_key_id = "qwreerwerwe"
+    }
+    samlp {
+      audience = "https://example.com/saml"
+      mappings = {
+        email = "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress"
+        name = "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name"
+      }
+      create_upn_claim = false
+      passthrough_claims_with_no_mapping = false
+      map_unknown_claims_as_is = false
+      map_identities = false
+      name_identifier_format = "urn:oasis:names:tc:SAML:2.0:nameid-format:persistent"
+      name_identifier_probes = [
+        "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress"
+      ]
+	  signing_cert = "-----BEGIN PUBLIC KEY-----\nMIGf...bpP/t3\n+JGNGIRMj1hF1rnb6QIDAQAB\n-----END PUBLIC KEY-----\n"
+    }
+  }
+  refresh_token {
+    leeway = 42
+    token_lifetime = 424242
+    rotation_type = "rotating"
+    expiration_type = "expiring"
+    infinite_token_lifetime = true
+    infinite_idle_token_lifetime = false
+    idle_token_lifetime = 3600
+  }
+  mobile {
+    ios {
+      team_id = "9JA89QQLNQ"
+      app_bundle_identifier = "com.my.bundle.id"
+    }
+  }
+  initiate_login_uri = "https://example.com/login"
+}
+`
+
+const testAccClientConfigWithoutAddonsWithSAMLPLogout = `
 resource "auth0_client" "my_client" {
   name = "Acceptance Test - {{.random}}"
   description = "Test Application Long Description"

